File: leetcode/python/1137_tribonacci_number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        dp = [0, 1, 1]
        if n < 3:
            return dp[n]
        for i in range(3, n+1):
            t = dp[0] + dp[1] + dp[2]
            dp[0], dp[1], dp[2] = dp[1], dp[2], t
        return dp[-1]

s = Solution()
print(s.tribonacci(4))
print(s.tribonacci(25))

File: leetcode/c++/1137_tribonacci_number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int dp[] = {0, 1, 1};
        if (n < 3) return dp[n];
        for (int i = 3; i <= n; i++) {
            int tmp = dp[0] + dp[1] + dp[2];
            dp[0] = dp[1];
            dp[1] = dp[2];
            dp[2] = tmp;
        }
        return dp[2];
    }
};